onnx/defs/math/defs.cc

```diff
@@ -1665,4 +1666,130 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
+void einsumRankInference(
+    ONNX_NAMESPACE::InferenceContext& ctx, std::string equation) {
+
+  const size_t numInputs = ctx.getNumInputs();
+  if (numInputs < 1 || !hasNInputShapes(ctx, static_cast<int>(numInputs))) {
+    return;
+  }
+
+  auto* output_shape = getOutputShape(ctx, 0);
+  std::string  left_equation;
+
+  equation.erase(std::remove(equation.begin(), equation.end(), ' '), equation.end()); // Remove whitespaces
```

Comment:
Whitespace generally includes other characters like tab and newline. I think it may be better to either use "isspace" function or change the documentation if only blank characters are permitted. I see that the numpy implementation is used as the reference implementation ... we could go with whatever numpy supports and document it.